### ISSUE TITLE
Update GCP Icons

### ISSUE THREAD

**Issue by soomineom** [mingrammer/diagrams#554]
Hi there,
I'm enjoying using Diagrams.

But I've noticed a lot of the GCP icons appear to be outdated... 

There is a new GCP architecture icons pack available at this link: https://cloud.google.com/icons
Is it possible to update the library to use these?

Thanks,
Soomin

**Comment by gabriel-tessier** [comment]
@soomineom 

Thanks for the link, if you feel comfortable with the contributions guide you can make a PR.

https://github.com/mingrammer/diagrams/blob/master/CONTRIBUTING.md

Or I can start checking about later this week.

**Comment by DataDavD** [comment]
hey @soomineom and @gabriel-tessier if y'all need help I'd be more than happy to assist in updating any icons for GCP or other cloud providers. Just let me know. I'll wait for soonmineom to confirm before starting though. 

I can take a look at other providers and see if they need updating too in mean time.

Happy coding

**Comment by soomineom** [comment]
@DataDavD Hi ,
I will really appreciate if u help me to update icons for GCP !

Thank you

**Comment by DataDavD** [comment]
Alright, sounds good! I should be able to get started/finish by end of this week🤞. I'll submit a PR or ask another question here if I have one.

cc @gabriel-tessier @soomineom .
Thank you!!

dd

**Comment by mingrammer** [comment]
Thank you guys :)

**Comment by DataDavD** [comment]
So, I took a look at the updated icons and I just wanna clarify that these updated ones appear to have removed the blue hexagon-shaped background for all the icons. Do you all think that is a particularly good idea for using those in diagrams? 

TL;DR: when updating diagrams what determines whether or not we should update? Is the purpose of this project to always use the most updated icons or those that look particular the best when used in complex diagrams. I ask because it seems like Draw.io is still using the icons with the blue hexagon-shaped background, but maybe they just take longer to update. I just wanna be clear what's best to use before working to add these. 

For reference I was referring to the .png file found in the `SVG and PNG Icons` set here. Thanks in advance for helping to clarify or feel free to ask more questions if my question was confusing

**Comment by soomineom** [comment]
This link (https://cloud.google.com/icons) says "During this transition period, we will not be updating the Powerpoint, Draw.IO, Lucidchart, SimpleDiagrams, or Cacoo resources.".

So I think we should think the most updated version is the icon without blue hexagon... 
Thank you

**Comment by DataDavD** [comment]
Ok thanks for confirm @soomineom, just wanted to make sure before I start working on this. I hope to get submit this PR by end of week, but I may have questions for you or @mingrammer before. 

Thanks again for your help and letting me help you on this awesome project!!!

dd

**Comment by DataDavD** [comment]
hey @mingrammer should I change the resource dirs to match the update icon set from GCP? I noticed the directories don't match the GCP resources directory's sub-directory names.

**Comment by DataDavD** [comment]
hey @mingrammer and @soomineom I was wondering if you were able to take a look at my question. I wanted to get clarification before I really start working on this. Thanks in advance!!!!

**Comment by mingrammer** [comment]
@DataDavD Sorry for the late reply.

Sure! I think we can switch to the most updated icons without hexagons. But, the updating process seems to be a work in progress now. So, you should work on it after the officially updated version is released.

And, if there are changes to the directory structure, we should release the next version with breaking changes release notes for existing users.

**Comment by mingrammer** [comment]
I assign this issue to you, @DataDavD.

**Comment by mingrammer** [comment]
Thank you for raising this issue too, @soomineom

**Comment by DataDavD** [comment]
No worries about the late reply!!! 

All sounds good. Thanks for clarifying that @mingrammer. The breaking changes was what I was worried about and was wondering how I should tackle it. 

One other question: when you say that the `updating process seems to be a work in progress now` you clearly mean the GCP Icon set update, correct? I assume so, but wanna make sure. 

Thank you for assigning me and letting me work on this great project!!!!

best,
dd

**Comment by DataDavD** [comment]
Hey @mingrammer I was just checking in to see if you were able to review my last comment. Thanks in advance.

**Comment by DataDavD** [comment]
FYI, I went through a few GCP docs and answered my own question :). I'll keep an eye out for it and once the GCP icon set is fully released I will submit a PR. Thank you

**Comment by mingrammer** [comment]
@DataDavD Thank you for keeping eye on this issue :)

**Comment by DataDavD** [comment]
@mingrammer FYI GCP has finalized their updated icon set. I will be gone over next few weeks for holidays, but I will work on getting the GCP icons updated as soon as I am back at the start of 2022. I will let you know the progress at the start of January.

**Comment by DataDavD** [comment]
hey @mingrammer sorry for the very late update. I've been extremely bogged down with work and life stuff (trying to buy a house and move states/cities, all while wrapping up a couple big projects at work), so I haven't really gotten to touch this. I should have more time this weekend (finally got a day off), so I will be trying to work on and wrapping this up this weekend. If I have questions I'll be sure to @ mention you with the corresponding code/branch. Thanks for your understanding and patience.

best,
dd

**Comment by DataDavD** [comment]
hey @mingrammer I'm going through the updated icons and unfortunately GCP doesn't really organize all the icons (each set of svg's and png's are in their own specific directory), so I'm trying how to determine the best way to organize the updated GCP Icons. I was starting with the list version of this GCP developer cheat sheet. What do you think of starting here? https://googlecloudcheatsheet.withgoogle.com

FYI this is going to be a a large, breaking change given how much they added (and didn't organize at the same time). 

Let me know what you think before I start organizing everything further just in case you think we should go a different route; possibly just replace the existing icons with their updated ones and then make another PR (that is the real breaking change) with all the new icons/services? Is that better?

best,
david
